packages/flutter/lib/src/widgets/scroll_simulation.dart

```diff
@@ -147,10 +152,9 @@ class ClampingScrollSimulation extends Simulation {
     required this.velocity,
```

Comment:
is the "some simplifications have been made" bit still true?

Reply:
Yes. The logic in `Scroller.java` is 2-dimensional and has logic for non "fling" scrolls so there are simplifications still.